src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -153,6 +165,43 @@ internal ImmutableArray<FunctionMetadata> LoadFunctionMetadata(bool forceRefresh
             var functionMetadataList = new List<FunctionMetadata>();
             _functionErrors = new Dictionary<string, ICollection<string>>();
 
+            // If routeHandling.mode is "all", ignore any function metadata configured and expose a single implicit http-handler function.
+            // This ensures the FunctionMetadataManager is aware of the implicit handler so it can be resolved at runtime
+            // for logging/metrics and other metadata lookups.
+            if (string.Equals(_routeHandlingOptions?.Mode, "all", StringComparison.OrdinalIgnoreCase))
```

Comment:
It looks like this same logic is defined twice. Consider extracting it to an instance method on `RouteHandlingOptions` itself.